src/Guesser/AbstractTypeGuesser.php

```diff
@@ -31,7 +31,8 @@ protected function getParentMetadataForProperty($baseClass, $propertyFullName, M
             return $modelManager->getParentMetadataForProperty($baseClass, $propertyFullName);
         } catch (MappingException $e) {
             // no metadata not found.
-            return;
         }
+
+        return null;
```

Comment:
Why don't you let the return in the catch ?

Reply:
Because I think is more scalable to ensure `null` will be always returned by default, even if a new non-breaking `catch` block is added.

Reply:
Then, WDYT of this:
```
try {
    $metadata = $modelManager->getParentMetadataForProperty($baseClass, $propertyFullName);	            
} catch (MappingException $e) {
    // no metadata not found.
    $metadata = null;	
}

return $metadata;
```

Reply:
>Then, WDYT of this:

In my replacement that occurrence was not found since I basically search for "return;". I think this case is different since `try` and `catch` blocks are assigning a value to a variable which will be always returned.
Are you proposing to do the same here?

Reply:
Sorry if I was not clear.

I was proposing to write
```
try {
    $metadata = $modelManager->getParentMetadataForProperty($baseClass, $propertyFullName);	            
} catch (MappingException $e) {
    // no metadata not found.
    $metadata = null;	
}

return $metadata;
```
Instead of 
```
try {
    return $modelManager->getParentMetadataForProperty($baseClass, $propertyFullName);	            
} catch (MappingException $e) {
    // no metadata not found.
}

return null;
```

Reply:
Don't worry, that's what I understood :+1:

Reply:
> in case someone add more catch and forget to return a value.

We can setup PHPStan to detect that.

Reply:
Please, let me know if you prefer this change to be reverted.

Reply:
I made code suggestions below, apply them and you will have my approval.

Reply:
Great. These changes were reverted.